app/routes/api.js

```diff
@@ -169,4 +170,92 @@ router.post('/report', (req, res) => {
     res.redirect('back');
 });
 
+// API endpoint to delete an attachment for a given attachment ID
+router.delete('/attachment/:attachmentID', (req, res, next) => {
+    res.setHeader('Content-Type', 'application/json')
+    try {
+        const attachmentID = req.params.attachmentID;
+
+        const result = deleteAttachmentforFactoid(attachmentID)
+
+        if (result) {
+            return res.status(200).send({ message: 'Attachment deleted successfully.' });
+        } else {
+            return res.status(404).send({ message: 'Attachment not found.' });
+        }
+    } catch (e) {
+        console.log(e);
+        return res.status(500).send({ message: 'Server error.' });
+    }
+});
+
+// API endpoint to delete a tag for a given factoid ID and category ID
+router.delete('/tag/:factoidID/:categoryID', (req, res) => {
+    res.setHeader('Content-Type', 'application/json')
+    try {
+        const factoidID = req.params.factoidID;
+        const categoryID = req.params.categoryID;
+
+        const result = deleteTagforFactoid(parseInt(factoidID), parseInt(categoryID));
```

Comment:
I feel like we should do this instead
```suggestion
        const factoidID = parseInt(req.params.factoidID);
        const categoryID = parseInt(req.params.categoryID);

		if (isNaN(factoidID) || isNaN(categoryID)) {
			return res.status(404).send({ message: 'Invalid fact or category in request.' });
		}

        const result = deleteTagforFactoid(factoidID, categoryID);
```